flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_AirwaysFromWaypointPage.js

```diff
@@ -63,14 +69,15 @@ class A320_Neo_CDU_AirwaysFromWaypointPage {
                     rows[i] = [`${pendingAirway.name}[color]cyan`, "[\xa0\xa0\xa0][color]cyan"];
                     mcdu.onRightInput[i] = (value) => {
                         if (value.length > 0) {
-                            mcdu.ensureCurrentFlightPlanIsTemporary(() => {
-                                mcdu.insertWaypointsAlongAirway(value, mcdu.flightPlanManager.getEnRouteWaypointsLastIndex() + 1, pendingAirway.name, (result) => {
+                            fmc.ensureCurrentFlightPlanIsTemporary(() => {
+                                fmc.insertWaypointsAlongAirway(value, fmc.flightPlanManager.getEnRouteWaypointsLastIndex() + 1, pendingAirway.name, (result) => {
                                     if (result) {
-                                        A320_Neo_CDU_AirwaysFromWaypointPage.ShowPage(mcdu, waypoint, offset);
+                                        A320_Neo_CDU_AirwaysFromWaypointPage.ShowPage(fmc, mcdu, waypoint, offset);
                                     } else {
                                         mcdu.addNewMessage(NXSystemMessages.awyWptMismatch);
                                     }
                                 });
+                                mcdu.requestOffsideUpdate();
```

Comment:
**issue**: `mcdu.requestOffsideUpdate();` may execute before `fmc.insertWaypointsAlongAirway` calls the callback, due to the asynchronous nature of `fmc.insertWaypointsAlongAirway`.
**suggestion**: Either await `insertWaypointsAlongAirway` or move the `requestOffsideUpdate` into the callback.